samples/azure-quantum/shorter-quantum-circuits/shorter-quantum-circuits-dataset.ipynb

```diff
@@ -0,0 +1,782 @@
+    "        print(member.name)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### File structure\n",
+    "\n",
+    "Each file contains an array $a_k$ of the best found approximation protocols that use a denominator power of at most $k$ in the branch that is applied with the highest probability. Each element of the array corresponds to one instance of the approximation protocol.\n",
+    "\n",
+    "Each protocol instance description in the JSON file contains three keys:\n",
+    "\n",
+    "* `target_angle` for the target rotation angle\n",
+    "* `protocol_name` for the protocol name\n",
+    "* `elements` for a description of the gate sequences and the probabilities with which these sequences are applied."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tarfile\n",
+    "import json\n",
+    "with tarfile.open('clifford-root-t-fourier-angles.tar.gz') as tar:\n",
+    "    protocol_instances = json.loads(tar.extractfile(tar.getmembers()[2]).read().decode())\n",
+    "    print(type(protocol_instances), ' of length ', len(protocol_instances))\n",
+    "    print(\"Example of a protocol instance:\\n\", json.dumps(protocol_instances[-1],indent=2))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Protocol instance structure\n",
+    "\n",
+    "The array corresponding to the key `element` of the protocol instance data structure is an array of length one, two, four or six. \n",
+    "Each array item contains three keys:\n",
+    "* `gates` - a string representing a product of unitary matrices $V$, for example `Rz[1,16].H.S` represents matrix product $V = \\exp(-\\frac{i\\pi}{16}Z)HS$\n",
+    "* `apply_as` - describes how the matrix product $V$ above is used in the protocol, one of `unitary`, `twirl` or `projection`\n",
+    "* `probability` - $p$, floating point number between zero and one, the probability with which `gates` will be applied.\n",
+    "\n",
+    "When `apply_as` is `unitary`, the product unitary $V$ directly applied to the target qubit on which rotation is to be executed.\n",
+    "When `apply_as` is `twirl`, one of the four unitaries $V$, $SVS^\\dagger$, $ZVZ$, $(SZ)V(SZ)^\\dagger$ \n",
+    "is applied each with probability $p/4$. \n",
+    "When `apply_as` is `projection`, the product unitary $V$ is applied to an ancillary qubit as a part of projective rotation circuit in \n",
+    "[Figure 7](https://arxiv.org/pdf/2203.10064v1.pdf#page=19) in the paper shown below.\n",
+    "\n",
+    "The projective rotation circuit is applied with probability $p$.\n",
+    "Next we discuss some protocol specific details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from IPython.display import Image\n",
```

Comment:
Instead of using a code cell you can include the image into Markdown using base64 encoding, see, e.g., here: https://github.com/microsoft/Quantum/blob/main/samples/azure-quantum/resource-estimation/estimation-chemistry.ipynb